src/geocel/GeantGeoParams.cc

```diff
@@ -383,6 +383,65 @@ std::vector<inp::Surface> make_inp_surfaces(GeantGeoParams const& geo)
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Create sensitive detectors input from Geant4 sensitive detectors.
+ */
+std::vector<inp::Detector> make_inp_detectors(GeantGeoParams const& geo)
+{
+    // Process detectors
+    std::vector<inp::Detector> result;
+
+    auto const& vol_labels = geo.impl_volumes();
+
+    // Process each logical volume
+    for (auto iv_id : range(ImplVolumeId{vol_labels.size()}))
+    {
+        auto const& label = vol_labels.at(iv_id);
+        if (label.empty())
+        {
+            // This volume isn't part of the world hierarchy
+            continue;
+        }
+
+        auto vol_id = geo.volume_id(iv_id);
+        auto& g4lv = *geo.id_to_geant(vol_id);
+
+        G4VSensitiveDetector* sd = g4lv.GetSensitiveDetector();
+        // check if volume is part of a sensitive detector
+        if (sd)
+        {
+            Label sd_label = sd->GetName();
+            bool existing_sd = false;
+
+            // Loop over existing input detectors in the result vector and
+            // append volume id to input detector volume id list if labels
+            // match.
+            for (auto detector : result)
```

Comment:
This is why the extra volumes weren't getting appended: you're unintentionally creating copies that get modified and discarded 🙃 
```suggestion
            for (auto& detector : result)
```